File: deck/seed/code/native/kotlin/runtime/path.kt
import java.io.File

// Path manipulation over java.io.File. The file extension carries its leading dot.
object path {
    fun join(base: String, name: String): String = File(base, name).path
    fun directory(target: String): String = File(target).parent ?: ""
    fun fileName(target: String): String = File(target).name
    fun fileExtension(target: String): String {
        val value = File(target).extension
        return if (value.isEmpty()) "" else ".$value"
    }
    fun isAbsolute(target: String): Boolean = File(target).isAbsolute
}

File: deck/seed/code/native/rust/runtime/path.rs
mod path {
    use std::path::Path;
    pub fn join(base: String, name: String) -> String {
        Path::new(&base).join(&name).to_string_lossy().to_string()
    }
    pub fn directory(target: String) -> String {
        Path::new(&target).parent().map(|p| p.to_string_lossy().to_string()).unwrap_or_default()
    }
    pub fn file_name(target: String) -> String {
        Path::new(&target).file_name().map(|p| p.to_string_lossy().to_string()).unwrap_or_default()
    }
    pub fn file_extension(target: String) -> String {
        Path::new(&target).extension().map(|p| format!(".{}", p.to_string_lossy())).unwrap_or_default()
    }
    pub fn is_absolute(target: String) -> bool {
        Path::new(&target).is_absolute()
    }
}
